docs/query.md

```diff
@@ -0,0 +1,265 @@
+| `branch:`    | `b:`    | Text                   | Searches within a specific branch.                         | `branch:main`                          |
+| `type:`      | `t:`    | `filematch`, `filename`, `file`, or `repo` | Limits result types.                   | `type:filematch`                       |
+
+---
+
+### 2. **Negation**
+
+Negate an expression using the `-` symbol.
+
+#### Examples:
+- Exclude a repository:
+  ```plaintext
+  -repo:"github.com/example/repo"
+  ```
+- Exclude a language:
+  ```plaintext
+  -lang:javascript
+  ```
+
+---
+
+### 3. **Grouping**
+
+Group queries using parentheses `()` to create complex logic.
+
+#### Examples:
+- Match either of two repositories:
+  ```plaintext
+  (repo:repo1 or repo:repo2)
+  ```
+- Find test in either python or javascript files:
+  ```plaintext
+  content:test (lang:python or lang:javascript)
+  ```
+
+---
+
+### 4. **Logical Operators**
+
+Use `or` to combine multiple expressions.
+
+#### Examples:
+- Match files in either of two languages:
+  ```plaintext
+  lang:go or lang:java
+  ```
+
+`and` boolean operator is applied automatically when expressions are separated by a space.
+
+---
+
+## Special Query Values
+
+- **Boolean Values**:
+  Use `yes` or `no` for fields like `archived:` or `fork:`.
+
+- **Text Fields**:
+  Text fields (`content:`, `repo:`, etc.) accept:
+  - Strings: `"my text"`
+  - Regular expressions: `/my.*regex/`
+
+- **Escape Characters**:
+  To include special characters, use backslashes (`\`).
+
+#### Examples:
+- Match the string `foo"bar`:
+  ```plaintext
+  content:"foo\"bar"
+  ```
+- Match the regex `foo.*bar`:
+  ```plaintext
+  content:/foo.*bar/
+  ```
+
+---
+
+## Advanced Examples
+
+1. **Search for content in Python files in public repositories**:
+   ```plaintext
+   lang:python public:yes content:"my_function"
+   ```
+
+2. **Exclude archived repositories and match a regex**:
+   ```plaintext
+   archived:no regex:/error.*handler/
+   ```
+
+3. **Find files named `README.md` in forks**:
+   ```plaintext
+   file:"README.md" fork:yes
+   ```
+
+4. **Search for a specific branch**:
+   ```plaintext
+   branch:main content:"TODO"
+   ```
+
+5. **Combine multiple fields**:
+   ```plaintext
+   (repo:"github.com/example" or repo:"github.com/test") and lang:go
+   ```
+
+---
+
+## Tips
+
+1. **Combine Filters**: You can combine as many fields as needed. For instance:
+   ```plaintext
+   repo:"github.com/example" lang:go content:"init"
+   ```
+
+2. **Use Regular Expressions**: Make complex content searches more powerful:
+   ```plaintext
+   content:/func\s+\w+\s*\(/
+   ```
+
+3. **Case Sensitivity**: Use `case:yes` for exact matches:
+   ```plaintext
+   case:yes content:"ExactMatch"
+   ```
+
+4. **Match Specific File Types**:
+   ```plaintext
+   file:".*\.go" content:"package main"
+   ```
+
+### EBNF Summary
+
+```ebnf
+query       = expression , { "or" , expression } ;
```

Comment:
nit: does this encode the fact that expressions can also be separated by spaces?

Maybe we want something like this:
```
query = expression, { separator, expression } ;
seperator = " " | "or"
```

Reply:
I think I'll omit this for now, then write a lezer grammar for zoekt query language and integrate it into the search bar.